### ISSUE TITLE
Vimtex 2.0 Math Zone

### ISSUE THREAD

**Issue by yushangakki** [lervag/vimtex#1870]
Hi Thanks for the big update! However, my old configuration on MathZone broke:

**.config/nvim/after/tex.vim**
```vim
syn clear texMathZoneW
syn region texMathZoneW	matchgroup=Delimiter start="\\\["			matchgroup=Delimiter	end="\\]\|%stopzone\>"			keepend contains=@texMathZoneGroup
call TexNewMathZone("E","align",1)
call TexNewMathZone("F","alignat",1)
call TexNewMathZone("G","equation",1)
call TexNewMathZone("H","flalign",1)
call TexNewMathZone("I","gather",1)
call TexNewMathZone("J","multline",1)
call TexNewMathZone("K","xalignat",1)
call TexNewMathZone("L","dmath",1)
```

**minimal.tex**
```tex
\documentclass{minimal}
\begin{document}
Hello world!
\end{document}
```

**Observed Behaviour**
No such highlight group name: texMathZoneW.

Unknown function: TexNewMathZone

**Expected Behaviour**
Make the align environment also be treated as the math zone.

Thanks! Happy Thanksgiving!

**Comment by lervag** [comment]
Your configuration is not necessary, because Vimtex already has good suppor for the amsmath package.:)

Happy thanksgiving!

**Comment by lervag** [comment]
Note that the `texMathZone*` groups do not exist, so no need for the `syn clear`. If you want to make new math zones, the method is called `vimtex#syntax#core#new_region_math`. An example:

```vim
call vimtex#syntax#core#new_region_math('gather')
```

Note, though, as said before: You don't need to do this, because Vimtex already adds these groups when you use the amsmath package.

**Comment by yushangakki** [comment]
Hi Dr. Lervag thanks for the quick reply, this question is actually about `Ultisnips`, where the context(math) breaks on Vimtex2.0

tex.snippets
```
global !p
texMathZones = ['texMathZone'+x for x in ['A', 'AS', 'B', 'BS', 'C', 'CS', 'D', 'DS', 'E', 'ES', 'F', 'FS', 'G', 'GS', 'H', 'HS', 'I', 'IS', 'J', 'JS', 'K', 'KS', 'L', 'LS', 'DS', 'V', 'W', 'X', 'Y', 'Z']]
texIgnoreMathZones = ['texMathText']
texMathZoneIds = vim.eval('map('+str(texMathZones)+", 'hlID(v:val)')")
texIgnoreMathZoneIds = vim.eval('map('+str(texIgnoreMathZones)+", 'hlID(v:val)')")
ignore = texIgnoreMathZoneIds[0]
def math():
	synstackids = vim.eval("synstack(line('.'), col('.') - (col('.')>=2 ? 1 : 0))")
	try:
		first = next(i for i in reversed(synstackids) if i in texIgnoreMathZoneIds or i in texMathZoneIds)
		return first != ignore
	except StopIteration:
		return False
endglobal

context "math()"
snippet ,ff "Fraction" iA
\frac{${1:numer}}{${2:denom}}${0}
endsnippet
```
After upgrading to vimtex 2.0, the snippets with trigger `,ff` doesn't work anymore.

May I get some insights on how I could resolve the issues here?

**Comment by lervag** [comment]
Yes, this was already reported and resolved some days ago, see #1863. Specifically, see [my comment](https://github.com/lervag/vimtex/issues/1863#issuecomment-733124368). :
